File: apps/mobile/ios/Visvine/Models/Account.swift
import Foundation

/// Authenticated user — mirrors src/types/index.ts `User`.
struct User: Codable, Identifiable, Equatable {
    let id: String
    var name: String
    var email: String
    var image: String?
    var nodeId: String?
    var isSuperAdmin: Bool?
}

/// GET /api/auth/session → { session: { user } | null }
struct SessionResponse: Codable {
    let session: SessionPayload?
}

struct SessionPayload: Codable {
    let user: User?
}

/// A seeded user offered by the dev-login path.
struct DevUser: Codable, Identifiable {
    let id: String
    let name: String
    let email: String
    let image: String?
}

struct DevUsersResponse: Codable {
    let users: [DevUser]
}

struct IssueTokenRequest: Codable {
    let userId: String
}

struct IssueTokenResponse: Codable {
    let token: String
    let user: DevUser
}

File: apps/mobile/android/app/src/main/java/com/visvine/mobile/data/model/Account.kt
package com.visvine.mobile.data.model

import kotlinx.serialization.Serializable

/** Authenticated user — mirrors src/types/index.ts `User`. */
@Serializable
data class User(
    val id: String,
    val name: String = "",
    val email: String = "",
    val image: String? = null,
    val nodeId: String? = null,
    val isSuperAdmin: Boolean? = null,
)

/** GET /api/auth/session → { session: { user } | null }. */
@Serializable
data class SessionResponse(
    val session: SessionPayload? = null,
)

@Serializable
data class SessionPayload(
    val user: User? = null,
)

/** A seeded user offered by the dev-login path (GET /api/dev/list-users). */
@Serializable
data class DevUser(
    val id: String,
    val name: String,
    val email: String,
    val image: String? = null,
)

@Serializable
data class DevUsersResponse(
    val users: List<DevUser> = emptyList(),
)

@Serializable
data class IssueTokenRequest(
    val userId: String,
)

@Serializable
data class IssueTokenResponse(
    val token: String,
    val user: DevUser,
)
